src/legacy/core_plugins/interpreter/public/functions/esaggs.ts

```diff
@@ -91,6 +100,14 @@ export const esaggs = (): ExpressionFunction<typeof name, Context, Arguments, Re
     const queryFilter = Private(FilterBarQueryFilterProvider);
 
     const aggConfigsState = JSON.parse(args.aggConfigs);
+    aggConfigsState.forEach(
```

Comment:
so we have this option on agg (so we could theoretically have two date histograms one with scaling off and one with scaling on) and then we also have this overall option on esaggs which will set it for every date_histogram ?

Reply:
should we skip the 'overall' option and just have this on date_histogram params ?

Reply:
(for now this means it will be just part of that json, but in the future we plan to add helper functions for generating agg configs (like `agg_date_histogram field='@timestamp' scale=false interval='1s'`)

Reply:
My concern was, that if we have this on `date_histogram` individually it's a) slightly trickier to make all old editor visualizations behave like that, without setting the default to `true` (which I don't think we want), and b) we create a potential new "feature" (different date histograms with different scaling settings), that we are not using anywhere or even planning to use, but that might fail later on.

For that reason I thought having only the `esaggs` flag, and not storing anything in date_histograms the easier solution for now. I only made it a date_histogram param, because I think that's currently the cleanest way to pass information from esaggs down to the AggConfig, but I would also be open for other suggestions for that.

What's your thoughts given that information?

Reply:
regarding b) for now we don't have any chart that would support multiple datehistograms so that option is just theoretical. seems confusing that we have an option there on esaggs that is only relevant for date_histogram.

regarding concern a) i agree its more complicated but this could be solved by migrating saved visualizations (on the fly in `vis_update_state.js`), when loading saved (legacy) vis, we would set this parameter to true. When creating new legacy visualizations in editor we would do the same (the old editor would just always set this option). opening already saved visualization would thus save in new format). for 8.0 we would need to migrate saved objects.